### PR TITLE
Merge latest upstream changes 

### PR DESCRIPTION
**Pull request by vocksel**
This PR merges the latest changes from commit 0d5efe33bc151377cbca86f6513190c6f4f6af29 in Roblox/jest-roblox

A new package JestMockGenv is included with these changes, and I'm hitting some build issues because of it. Seems like this package may need to be published to the jsdotlua npm scope first.

Checklist before submitting:
* [ ] Tidy up merge artifacts
* [ ] Added/updated relevant tests
* [ ] Added/updated documentation

### CODE REVIEW

File: src/jest-mock-genv/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@jsdotlua/jest-mock-genv",
+  "version": "3.6.1-rc.2",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/jsdotlua/jest-lua.git",
+    "directory": "src/jest-mock"
```
Comment (jeparlefrancais) [inline src/jest-mock-genv/package.json]:
```suggestion
    "directory": "src/jest-mock-genv"
```

**Review by jeparlefrancais — APPROVED** [review]
Minor thing! Other than this it looks good to me 👍 

Thanks a lot this is very valuable work 🎉

**Comment by MayGo** [comment]
Would really love if this get merged. Want to run it in Open Cloud Engine: https://devforum.roblox.com/t/beta-open-cloud-engine-api-for-executing-luau/3172185/67

**Comment by grilme99** [comment]
I'm going to go ahead and merge this despite the failing tests. I combed through and the test failures are due to the change in file structure in our fork, and aren't actually broken functionality. I don't have the time to fix the tests and snapshots, but need to get this merged in ASAP.

**Pull request merged**